qa/acceptance-tests/src/test/java/io/camunda/it/client/RoleIntegrationTest.java

```diff
@@ -104,11 +117,184 @@ void shouldRejectCreationIfMissingRoleName() {
   }
 
   @Test
-  void shouldReturnNotFoundIfRoleDoesNotExist() {
+  void shouldReturnNotFoundWhenGetRoleDoesNotExist() {
     // when / then
     assertThatThrownBy(() -> camundaClient.newRoleGetRequest("someRoleId").send().join())
         .isInstanceOf(ProblemException.class)
         .hasMessageContaining("Failed with code 404: 'Not Found'")
         .hasMessageContaining("Role with role ID someRoleId not found");
   }
+
+  @Test
+  void shouldRejectGetRoleIfNullRoleId() {
+    // when / then
+    assertThatThrownBy(() -> camundaClient.newRoleGetRequest(null).send().join())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining("roleId must not be null");
+  }
+
+  @Test
+  void shouldRejectGetRoleIfEmptyRoleId() {
+    // when / then
+    assertThatThrownBy(() -> camundaClient.newRoleGetRequest("").send().join())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining("roleId must not be empty");
+  }
+
+  @Test
+  void shouldDeleteRoleById() {
+    // given
+    camundaClient
+        .newCreateRoleCommand()
+        .roleId(ROLE_ID_3)
+        .name(ROLE_NAME_3)
+        .description(DESCRIPTION)
+        .send()
+        .join();
+
+    assertRoleCreated(ROLE_ID_3, ROLE_NAME_3, DESCRIPTION);
+
+    // when
+    camundaClient.newDeleteRoleCommand(ROLE_ID_3).send().join();
+
+    // then
+    Awaitility.await("Role is deleted")
+        .untilAsserted(
+            () ->
+                assertThatThrownBy(() -> camundaClient.newRoleGetRequest(ROLE_ID_3).send().join())
+                    .isInstanceOf(ProblemException.class)
+                    .hasMessageContaining("Failed with code 404: 'Not Found'"));
+  }
+
+  @Test
+  void shouldReturnNotFoundWhenRoleIdDoesNotExist() {
```

Comment:
```suggestion
  void shouldReturnNotFoundOnDeleteWhenRoleDoesNotExist() {
```

🔧 This makes it a bit more specific on what we're doing. Otherwise there's like 10 commands that could have the same method name 😄